core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -196,12 +196,15 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
 
         _removeCollateral(positionData.rawCollateral, collateralAmount);
         require(_checkPositionCollateralization(positionData));
-        _removeCollateral(rawTotalPositionCollateral, collateralAmount);
+        FixedPoint.Unsigned memory amountWithdrawn = _removeCollateral(rawTotalPositionCollateral, collateralAmount);
 
         // Move collateral currency from contract to sender.
-        collateralCurrency.safeTransfer(msg.sender, collateralAmount.rawValue);
+        // Note that we move the amount of collateral that is decreased from rawCollateral (inclusive of fees)
+        // instead of the user requested amount. This eliminates precision loss that could occur
+        // where the user withdraws more collateral than rawCollateral is decremented by.
```

Comment:
nit: can you add a note about why it only matters for the global collateral and not the position's collateral? I think our reasoning is that we need to ensure our bookkeeping around the global collateral is always <= the true amount of collateral in the contract to avoid reverts on withdrawals.

Reply:
done